src/app/public/modules/autonumeric/autonumeric.directive.ts

```diff
@@ -82,6 +82,12 @@ export class SkyAutonumericDirective implements OnInit, ControlValueAccessor, Va
   }
 
   public writeValue(value: number): void {
+
+    if (typeof value !== 'number' && value !== null && value !== undefined) {
+      console.warn(`'${value}' is invalid. The value provided to the autonumeric directive form control must be numeric.`);
```

Comment:
What's the value in adding a `console.warn` for an invalid value? Why not invalidate the field outright?

It seems that a user would not be able to add a non-numeric value by using the keyboard, so this would only be a problem if the initial value of the model is non-numeric?

Reply:
Yeah, that's not a bad point. You're correct -the only way to set the value this way would be through some programmatic logic.

Reply:
@Blackbaud-SteveBrush Ok, I moved the logic around a bit to now be a legit validator. If a value that's not `typeof` `number` is provided, it will add a `ValidationError` to the control. I also removed the console warning because now that just seems redundant.